src/MongoDB.Driver.Core/Core/Configuration/ClusterBuilderExtensions.cs

```diff
@@ -161,15 +167,20 @@ public static ClusterBuilder ConfigureWithConnectionString(this ClusterBuilder b
             // Server
 
             // Cluster
-            builder = builder.ConfigureCluster(s => s.With(connectionMode: connectionString.Connect));
+#pragma warning disable 618
+            builder = builder.ConfigureCluster(s => s.WithConnection(connectionMode: connectionString.Connect, directConnection: connectionString.DirectConnection));
```

Comment:
I've added a new obsolete method WithConnection to handle setting ConnectionMode and DirectConnection. It's a bit difficult to do it with the already existed With since ConnectionMode is not nullable.